WalletWasabi.Fluent/ViewModels/Wallets/Advanced/WalletCoins/WalletCoinsViewModel.cs

```diff
@@ -107,15 +106,11 @@ protected override void OnNavigatedTo(bool isInHistory, CompositeDisposable disp
 	{
 		base.OnNavigatedTo(isInHistory, disposables);
 
-		Observable.Merge(
-			_balanceChanged.Select(_ => Unit.Default),
-			_walletViewModel.WhenAnyValue(w => w.IsCoinJoining).Select(_ => Unit.Default))
-			.Subscribe(_ =>
-			{
-				Update();
-			});
-
-		disposables.Add(_coinsSourceList);
+		disposables.Add(_balanceChanged
+			.Select(_ => Unit.Default)
+			.Merge(_walletViewModel.WhenAnyValue(w => w.IsCoinJoining)
+			.Select(_ => Unit.Default))
+			.Subscribe(_ => Update()));
```

Comment:
When we `Subscribe` to changes in a property of foreign object, we get a reference to an IDisposable (that represents a subscription). We need to dispose this subscription properly when we no longer need it, in order to save resources.

This can be done by explicitly calling the `Dispose` method on the subscription. 

What @danwalmsley did was to add the subscription to a (provided) list of subscriptions, **disposables**) to let the navigation mechanism dispose the subscriptions appropriately.

This mechanism automates the disposal of any disposable as part of the navigation. 
1. When your ViewModel is navigated to (entering), you create your subscriptions and add them to the **disposables** list.
2. After your ViewModel is navigated from (leaving), the mechanism will dispose all the disposables added in step 1.